### PR TITLE
feat(material): add materials

### PR DESCRIPTION
**Pull request by SirXuNian**
add renderer base pass
add materials

### CODE REVIEW

File: src/engine/materials/GlassMaterial.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import { Vector4 } from '../math/Vector4';
+import { defaultRes } from '../textures/DefaultRes';
+import { PhysicMaterial } from './PhysicMaterial';
+/**
+ * GlassMaterial
+ * an rendering material implemented by simulating glass surfaces
+ * @group Material
+ */
+export class GlassMaterial extends PhysicMaterial {
+<<<<<<< HEAD
```
Comment (hellmor) [inline src/engine/materials/GlassMaterial.ts:14]: need to fix the conflict

File: src/engine/materials/GlassMaterial.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+=======
+    /**
+     * 实例个数
```
Comment (hellmor) [inline src/engine/materials/GlassMaterial.ts:17]: remove it?

File: src/engine/materials/GlassMaterial.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        ret.alphaCutoff = this.alphaCutoff;
+        ret.ior = this.ior;
+        ret.clearcoatFactor = this.clearcoatFactor;
+        ret.clearcoatRoughnessFactor = this.clearcoatRoughnessFactor;
+        return ret as this;
+    }
+
+<<<<<<< HEAD
+
+=======
```
Comment (hellmor) [inline src/engine/materials/GlassMaterial.ts:88]: fix the conflict

File: src/engine/materials/GlassMaterial.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    /**
+     * 启用GUI调试
```
Comment (hellmor) [inline src/engine/materials/GlassMaterial.ts:90]: remove it

**Review by Codeboy-cn — APPROVED** [review]

**Pull request merged**